drv/ignition-api/build.rs

```diff
@@ -2,10 +2,22 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
+use build_fpga_regmap::fpga_regs;
+use std::{fs, io::Write};
+
 fn main() -> Result<(), Box<dyn std::error::Error>> {
     idol::client::build_client_stub(
         "../../idl/ignition.idol",
         "client_stub.rs",
     )?;
+
+    let out_dir = build_util::out_dir();
+    let mut reg_map = fs::File::create(out_dir.join("ignition_controller.rs"))?;
+    write!(
+        &mut reg_map,
+        "{}",
+        fpga_regs(include_str!("ignition_controller.json"))?,
```

Comment:
Tiny nit - is `include_str!()` necessary? It's probably fine, but I've had issues with compiler times / memory usage in the past with the `include_*!()` macros, and since this is a build script, we could probably do `std::fs::read_to_string()` instead.